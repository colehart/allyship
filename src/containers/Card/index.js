import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { toggleSaved } from '../../actions';
import './Card.css';

export const Card = (props) => {
  const {
    source,
    author,
    title,
    description,
    url,
    urlToImage,
    published,
    isSaved,
    toggleSaved
  } = props

  return (
    <article className='Card'>
      <div className='c-title-group'>
        <h3 className='c-title'>{title}</h3>
        <div
          className={`c-save ${ isSaved ? 'is-saved' : '' }`}
          onClick={() => toggleSaved(title)}
        >
        </div>
      </div>
      <p className='c-byline'>{published} | {source} {`${author ? `| ${author}` : ''}`}</p>
      <img className='c-image' src={urlToImage} alt='provided illustration'/>
      <p className='c-description'>{description}</p>
      <a
        className='c-link'
        href={url}
        target='_blank'
        rel='noopener noreferrer'
      >
        Read article from source
      </a>
    </article>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  toggleSaved: (headline) => dispatch(toggleSaved(headline))
})

Card.propTypes = {
  toggleSaved: PropTypes.func
}

export default connect(null, mapDispatchToProps)(Card)