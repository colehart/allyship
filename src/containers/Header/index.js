import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import shuttleIcon from '../../assets/images/space-shuttle.svg';
import './Header.css';

export const Header = () => {
  return (
    <div className='Header'>
      <Link
        to='/'
        className='h-logo-group'
      >
        <img
          className='h-logo'
          src={ shuttleIcon }
          alt='A space shuttle'
        />
        <h1>Allyship</h1>
      </Link>
      <nav className='h-nav-group'>
        <NavLink
          exact to='/'
          className="h-categories btn"
        >
          Categories
        </NavLink>
        <NavLink
          exact to='/saved'
          className="h-saved btn"
        >
          Saved
          <span className='h-badge'> 0</span>
        </NavLink>
      </nav>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  caughtError: state.caughtError,
  stories: state.stories
})

export default connect(mapStateToProps)(Header)