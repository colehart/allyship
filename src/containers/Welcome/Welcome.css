.Welcome {
  width: 75%;
  max-width: 780px;
  margin: 0 auto;
  grid-row: 2 / 3;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.w-message,
.w-category-group {
  padding: 40px 0;
  max-width: 550px;
}

.w-message h2 {
  text-align: center;
}

.w-message p {
  font-size: 1.15rem;
}

.w-category-group {
  display: flex;
  justify-content: space-between;
}

@media screen and (max-width: 655px) {
  .w-category-group {
    height: 215px;
    flex-direction: column;
    align-items: center;
  }

  .w-category-group .btn {
    width: 100%;
  }
}