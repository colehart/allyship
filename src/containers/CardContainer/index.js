import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import Card from '../Card';
import './CardContainer.css';

export const CardContainer = (props) => {
  const { isLoading } = props
  const cards = []

  return (
    <main className='CardContainer'>
      { isLoading ? <Loader /> : null }
      { cards }
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

CardContainer.propTypes = {
  isLoading: PropTypes.bool.isRequired
}

export default connect(mapStateToProps)(CardContainer)