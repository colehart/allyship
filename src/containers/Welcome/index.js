import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import { fetchStories } from '../../thunks/fetchStories';
import './Welcome.css';

export const Welcome = (props) => {
  const { isLoading } = props

  return (
    <main className='Welcome'>
      { isLoading ? <Loader /> : null }
      <section
        className={ isLoading ? 'hidden' : 'w-message'}
      >
        <h2>Welcome!</h2>
        <p>
          Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
        <br />
        <br />
          Research the latest news about a specific minority identity or issue.
        {/* , or choose your own query by entering a search keyword or phrase. */}
        </p>
      </section>
      <div
        className='w-category-group'
      >
        <Link
          to='/transgender'
          exact='true'
          className='btn'
          aria-label='Click here for the latest transgender news'
        >
          Transgender
        </Link>
        <Link
          to='/immigration'
          exact='true'
          className='btn'
          aria-label='Click here for the latest immigration news'
        >
          Immigration
        </Link>
        <Link
          to='/black-lives-matter'
          exact='true'
          aria-label='Click here for the latest black lives matter news'
          className='btn'
        >
          Black Lives Matter
        </Link>
      </div>
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchStories: (url) => dispatch(fetchStories(url))
})

Welcome.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  fetchStories: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
