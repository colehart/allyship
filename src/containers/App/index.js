import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Header } from '../Header';
import { Welcome } from '../Welcome';
import { CardContainer } from '../CardContainer';
import './App.css';

export const App = (props) => {
  const { isLoading } = props

  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Your ship has gone off course.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the orange buttons in this app to navigate to your desired destination.</p>
    </div>
  )

  return (
    <div className='App'>
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ Welcome }
        />
        <Route
          exact path='/saved'
          render={() => CardContainer }
        />
        <Route
          exact path='/transgender'
          render={() => CardContainer }
        />
        <Route
          exact path='/immigration'
          render={() => CardContainer }
        />
        <Route
          exact path='/black-lives-matter'
          render={() => CardContainer }
        />
        <Route
          component={noMatch}
        />
      </Switch>
      {/* <Route
        exact path='/search'
        render={() => CardContainer }
      /> */}
    </div>
  );
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
})

export default connect(mapStateToProps)(App);
