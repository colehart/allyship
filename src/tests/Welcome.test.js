import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Welcome, mapStateToProps } from '../containers/Welcome';
import { mockDefaultState, mockFullState } from './testMocks'

describe('Welcome', () => {
  describe('Welcome Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Welcome isLoading={false}/>)
    })

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should render the Loader if isLoading(true)', () => {
      wrapper = shallow(<Welcome isLoading={true} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should parse isLoading from state', () => {
      const expected1 = false

      const mappedProps1 = mapStateToProps(mockDefaultState)
      expect(mappedProps1.isLoading).toEqual(expected1)

      const expected2 = true

      const mappedProps2 = mapStateToProps(mockFullState)
      expect(mappedProps2.isLoading).toEqual(expected2)
    })
  })
})