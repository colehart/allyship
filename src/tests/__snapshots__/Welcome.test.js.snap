// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Welcome Component matches the snapshot 1`] = `
<main
  className="Welcome"
>
  <section
    className="w-message"
  >
    <h2>
      Welcome!
    </h2>
    <p>
      Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
      <br />
      <br />
      Research the latest news about a specific minority identity or issue.
    </p>
  </section>
  <div
    className="w-category-group"
  >
    <Link
      aria-label="Click here for the latest transgender news"
      className="btn"
      exact="true"
      replace={false}
      to="/transgender"
    >
      Transgender
    </Link>
    <Link
      aria-label="Click here for the latest immigration news"
      className="btn"
      exact="true"
      replace={false}
      to="/immigration"
    >
      Immigration
    </Link>
    <Link
      aria-label="Click here for the latest black lives matter news"
      className="btn"
      exact="true"
      replace={false}
      to="/black-lives-matter"
    >
      Black Lives Matter
    </Link>
  </div>
</main>
`;

exports[`Welcome Welcome Component should render the Loader if isLoading(true) 1`] = `
<main
  className="Welcome"
>
  <Loader />
  <section
    className="hidden"
  >
    <h2>
      Welcome!
    </h2>
    <p>
      Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
      <br />
      <br />
      Research the latest news about a specific minority identity or issue.
    </p>
  </section>
  <div
    className="w-category-group"
  >
    <Link
      aria-label="Click here for the latest transgender news"
      className="btn"
      exact="true"
      replace={false}
      to="/transgender"
    >
      Transgender
    </Link>
    <Link
      aria-label="Click here for the latest immigration news"
      className="btn"
      exact="true"
      replace={false}
      to="/immigration"
    >
      Immigration
    </Link>
    <Link
      aria-label="Click here for the latest black lives matter news"
      className="btn"
      exact="true"
      replace={false}
      to="/black-lives-matter"
    >
      Black Lives Matter
    </Link>
  </div>
</main>
`;
